src/string_buffer.rs

```diff
@@ -134,9 +155,18 @@ impl StringNode {
         }
     }
 
-    // None if there is no new line in this node.
+    // Returns the length of the string stored in the list starting in this node
+    fn len(&self) -> usize {
```

Comment:
I would give this a more explanatory name - len() to me suggests self.data.len(), rather than what is described in the comment.